From 7672dd73d40f7e33a9b7fb274b7c760a4c0cb89e Mon Sep 17 00:00:00 2001
From: Even Rouault <even.rouault@spatialys.com>
Date: Wed, 21 May 2025 13:59:13 +0200
Subject: [PATCH] Python bindings: fix build error on Windows with setuptools
 80.0

---
 swig/python/setup.py.in | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/swig/python/setup.py.in b/swig/python/setup.py.in
index 547ea70ed699..75990f415f02 100644
--- a/swig/python/setup.py.in
+++ b/swig/python/setup.py.in
@@ -12,6 +12,7 @@ import sys
 from glob import glob
 from pathlib import Path
 
+import setuptools
 from setuptools.command.build_ext import build_ext
 from setuptools import setup
 from setuptools import find_packages
@@ -157,7 +158,7 @@ def has_flag(compiler, flagname):
 # since they may be directly used by users. So we do a monkey patching of
 # easy_install.install_wrapper_scripts to install a modified
 # easy_install.delete_blockers method that does NOT remove .py files
-if sys.platform == 'win32':
+if sys.platform == 'win32' and int(setuptools.__version__.split('.')[0]) < 80:
     from setuptools.command.easy_install import easy_install
 
     original_install_wrapper_scripts = easy_install.install_wrapper_scripts
