diff --git a/swig/python/setup.py.in b/swig/python/setup.py.in
index e6e542c9f43c..51e08870ae41 100644
--- a/swig/python/setup.py.in
+++ b/swig/python/setup.py.in
@@ -364,34 +364,34 @@ extra_compile_args = []
 
 gdal_module = Extension('osgeo._gdal',
                         sources=['extensions/gdal_wrap.cpp'],
-                        extra_compile_args=extra_compile_args,
-                        extra_link_args=extra_link_args)
+                        extra_compile_args=list(extra_compile_args),
+                        extra_link_args=list(extra_link_args))
 
 gdalconst_module = Extension('osgeo._gdalconst',
                              sources=['extensions/gdalconst_wrap.c'],
-                             extra_compile_args=extra_compile_args,
-                             extra_link_args=extra_link_args)
+                             extra_compile_args=list(extra_compile_args),
+                             extra_link_args=list(extra_link_args))
 
 osr_module = Extension('osgeo._osr',
                        sources=['extensions/osr_wrap.cpp'],
-                       extra_compile_args=extra_compile_args,
-                       extra_link_args=extra_link_args)
+                       extra_compile_args=list(extra_compile_args),
+                       extra_link_args=list(extra_link_args))
 
 ogr_module = Extension('osgeo._ogr',
                        sources=['extensions/ogr_wrap.cpp'],
-                       extra_compile_args=extra_compile_args,
-                       extra_link_args=extra_link_args)
+                       extra_compile_args=list(extra_compile_args),
+                       extra_link_args=list(extra_link_args))
 
 
 array_module = Extension('osgeo._gdal_array',
                          sources=['extensions/gdal_array_wrap.cpp'],
-                         extra_compile_args=extra_compile_args,
-                         extra_link_args=extra_link_args)
+                         extra_compile_args=list(extra_compile_args),
+                         extra_link_args=list(extra_link_args))
 
 gnm_module = Extension('osgeo._gnm',
                        sources=['extensions/gnm_wrap.cpp'],
-                       extra_compile_args=extra_compile_args,
-                       extra_link_args=extra_link_args)
+                       extra_compile_args=list(extra_compile_args),
+                       extra_link_args=list(extra_link_args))
 
 ext_modules = [gdal_module,
                gdalconst_module,
