diff --git a/sipbuild/__init__.py b/sipbuild/__init__.py
index 3a3b094..a551b47 100644
--- a/sipbuild/__init__.py
+++ b/sipbuild/__init__.py
@@ -22,6 +22,8 @@
 
 
 # Publish the public API.
+import os
+os.add_dll_directory(os.path.abspath(os.path.join(__file__, "../../../../../../bin")))
 from .abstract_builder import AbstractBuilder
 from .abstract_project import AbstractProject
 from .bindings import Bindings
